pkg/controller/direct/certificatemanager/dnsauthorization_mappings.go

```diff
@@ -20,9 +20,14 @@ import (
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
 )
 
-func CertificateManagerDNSAuthorizationStatusObservedState_FromProto(mapCtx *direct.MapContext, created *certificatemanagerpb.DnsAuthorization) *krm.CertificateManagerDNSAuthorizationStatus {
-	status := &krm.CertificateManagerDNSAuthorizationStatus{
-		DnsResourceRecord: []krm.DnsAuthorization_DnsResourceRecord{direct.ValueOf(DnsAuthorization_DnsResourceRecord_FromProto(mapCtx, created.GetDnsResourceRecord()))},
+func CertificateManagerDNSAuthorizationStatusObservedState_FromProto(mapCtx *direct.MapContext, in *certificatemanagerpb.DnsAuthorization) *krm.CertificateManagerDNSAuthorizationStatus {
```

Comment:
You _should_ be able to add a `// +kcc:status:proto=google.cloud...` annotation and it _should_ come closer.  That said I normally just map these fields manually in an updateStatus function.